source/version_helper.js

```diff
@@ -10,5 +11,34 @@ module.exports = {
       // in that case we fallback to a branch version
       return semver.coerce(version === '0.0.0-development' ? branchVersion : version);
   },
+  CockroachVersionGreaterThan21_1: async function(connection) {
+    const versionRow = await connection.query("SELECT version() AS version", { type: QueryTypes.SELECT });
+    const cockroachDBVersion = versionRow[0]["version"]
+
+    // This regex pattern should match any version 21.1 and greater.
+    // Also assumes that the naming scheme will always follow v[XX].[Y].[ZZZZ]
+    // Where XX is the last two digits of the year, Y is the subrelease (currently is always 1/2) and
+    // ZZZZ is the point release version.
+    // This has to be updated if the version format changes or a new version of CockroachDB is released in the year 2100.
+    var re = new RegExp('v(2[1-9]|[3-9][0-9]).[1-2].[0-9]*')
+    return cockroachDBVersion.match(re)
+  },
+  CockroachVersionGreaterThan21_2: async function(connection) {
+    const versionRow = await connection.query("SELECT version() AS version", { type: QueryTypes.SELECT });
+    const cockroachDBVersion = versionRow[0]["version"]
+
+    // This regex pattern matches any 21.2 version.
+    var re21_2 = 'v21.2.[1-2].[0-9]*'
```

Comment:
what is the `[1-2]` part for? this looke like it would not match v21.2.3, which should be counted

also, should the periods be escaped?

Reply:
Oops yes this is incorrect, added extra [1-2] by accident